CRM/Report/Form.php

```diff
@@ -4909,7 +4909,12 @@ public function getLabels($options, $baoName, $fieldName) {
     $types = explode(',', $options);
     $labels = array();
     foreach ($types as $value) {
-      $labels[$value] = CRM_Core_PseudoConstant::getLabel($baoName, $fieldName, $value);
+      if ($labelValue = CRM_Core_PseudoConstant::getLabel($baoName, $fieldName, $value)) {
+        $labels[$value] = $labelValue;
+      }
+      else {
+        $labels[$value] = CRM_Core_DAO::getFieldValue($baoName, $value, $fieldName);
```

Comment:
I think if you passed in the BAO Name for EntityBatch & added the pseudoconstant to the xml for batch_id in entity_batch you would not need to change this function like this, & I think adding metadata to the xml generally is a good way to do things (although you will need to regen the DAOs)